src/Conversion/ONNXToKrnl/Math/LRN.cpp

```diff
@@ -55,87 +55,86 @@ struct ONNXLRNOpLowering : public ConversionPattern {
-        rewriter, loc);
-
-    Value sumAlloc = create.mem.alloc(scalarMemRefType);
-    create.krnl.store(emitConstantOp(rewriter, loc, elementType, 0), sumAlloc);
-
-    // Create the sum reduction loop
-    BuildKrnlLoop sumLoops(rewriter, loc, 1);
-    sumLoops.createDefineOp();
-    sumLoops.pushBounds(lbMaxList, ubMinList);
-    sumLoops.createIterateOp();
-    auto outputLoopBody = rewriter.saveInsertionPoint();
-    rewriter.setInsertionPointToStart(sumLoops.getIterateBlock());
-
-    // Compute quare-sum value
-    SmallVector<Value, 4> loadIndices;
-    for (int i = 0; i < outputRank; i++) {
-      if (i == loopIndexForC) {
-        Value loopVal = sumLoops.getInductionVar(0);
-        loadIndices.emplace_back(loopVal);
-      } else {
-        Value loopVal = outputLoops.getInductionVar(i);
-        loadIndices.emplace_back(loopVal);
-      }
-    }
-
-    Value loadVal = create.krnl.load(input, loadIndices);
-    Value squareVal = create.math.mul(loadVal, loadVal);
-
-    Value sumValue = create.krnl.load(sumAlloc, ArrayRef<Value>{});
-    sumValue = create.math.add(sumValue, squareVal);
-    create.krnl.store(sumValue, sumAlloc, ArrayRef<Value>{});
-
-    // Compute and store the output
-    // y = x / ((bias + (alpha / nsize) * square_sum) ** beta)
-    rewriter.restoreInsertionPoint(outputLoopBody);
-    SmallVector<Value, 4> storeIndices;
-    for (int i = 0; i < outputRank; ++i) {
-      storeIndices.emplace_back(outputLoops.getInductionVar(i));
-    }
-    Value xValue = create.krnl.load(input, storeIndices);
-    sumValue = create.krnl.load(sumAlloc);
-    Value tempValue =
-        create.math.pow(create.math.add(biasValue,
-                            create.math.mul(alphaDivSizeValue, sumValue)),
-            betaValue);
-    Value resultValue = create.math.div(xValue, tempValue);
-
-    create.krnl.store(resultValue, alloc, storeIndices);
+    KrnlBuilder createKrnl(rewriter, loc);
+    ValueRange outputLoopDef = createKrnl.defineLoops(outputRank);
+    SmallVector<IndexExpr, 4> lbs(outputRank, LiteralIndexExpr(0));
+    createKrnl.iterateIE(outputLoopDef, outputLoopDef, lbs,
+        shapeHelper.dimsForOutput(0),
+        [&](KrnlBuilder &createKrnl, ValueRange outputLoopInd) {
+          // Insert computation of square_sum.
+          // square_sum[n, c, d1, ..., dk] = sum(X[n, i, d1, ..., dk] ^ 2),
+          // where max(0, c - floor((size - 1) / 2)) <= i
+          // and i<= min(C - 1, c + ceil((size - 1) / 2)).
+
+          // Get a child IndexExpr context.
+          IndexExprScope childScope(&rewriter, shapeHelper.scope);
+
+          // Compute the lower bound and upper bound for square_sum.
+          const int loopIndexForC = 1;
+          DimIndexExpr cIE(outputLoopInd[loopIndexForC]);
+          MemRefBoundsIndexCapture inputBounds(input);
+          DimIndexExpr CIE(inputBounds.getDim(loopIndexForC));
+          SymbolIndexExpr sizeIE = LiteralIndexExpr(sizeLit);
```

Comment:
I'm wondering whether we could avoid the copy and write:
```
SymbolIndexExpr sizeIE(sizeLit);

Reply:
```
LiteralIndexExpr(int64_t const value);
```
`LiteralIndexExpr` has a constructor for `int`, but `SymbolIndexExpr` doesn't.